src/theme/Root.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from 'react'
+import { useLocation } from '@docusaurus/router'
+
+import { initializeAnalytics, sendAnalyticsEvent, Trace, user } from '@uniswap/analytics'
+import { CustomUserProperties, EventName, PageName } from '@uniswap/analytics-events'
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext'
+import { getCLS, getFCP, getFID, getLCP, Metric } from 'web-vitals'
+import { getBrowser } from '../utils/browser'
+
+// Placeholder API key. Actual API key used in the proxy server
+const ANALYTICS_DUMMY_KEY = '00000000000000000000000000000000'
+
+function getCurrentPageFromLocation(locationPathname: string): PageName | undefined | string {
+  if (locationPathname === '/') {
+    return PageName.HOME_PAGE
+  }
+  const pathWithoutInitialSlash = locationPathname.slice(1)
+  const pathWithSlashesReplaced = pathWithoutInitialSlash.replace(/\//g, '-')
```

Comment:
Are we concerned with ambiguity if a URL uses dashes? Is there a different character we can use here that's less likely to be used inside a word?

Reply:
Good call!